src/mango/src/mango_execution_stats.erl

```diff
@@ -82,9 +83,11 @@ log_end(Stats) ->
 
 
 maybe_add_stats(Opts, UserFun, Stats, UserAcc) ->
+    Stats0 = log_end(Stats),
```

Comment:
A variable with a 0 is lower that Stats. So you could make the first variable of Stats as Stats0 eg:

```
maybe_add_stats(Opts, UserFun, Stats0, UserAcc) ->
     Stats = log_end(Stats0), 
```
Is how its typically done in the CouchDB codebase